src/WebJobs.Extensions.ApiHub/Config/ApiHubConfiguration.cs

```diff
@@ -154,7 +154,25 @@ private Task<ApiHubFile> BuildFromAttribute(ApiHubFileAttribute attribute)
 
         private IFolderItem GetFileSource(string key)
         {
-            return _map[key];
+            if (_map.ContainsKey(key))
+            {
+                return _map[key];
+            }
+            else
+            {
+                // If the key doesn't exist, see if it is available as an environment variable.
+                // This might be the case for imperative binding scenarios.
+                var connectionString = Environment.GetEnvironmentVariable(key);
+                if (!string.IsNullOrWhiteSpace(connectionString))
+                {
+                    AddConnection(key, connectionString);
```

Comment:
>AddConnection [](start = 20, length = 13)

Is the bug that AddConnection is currently only called  when there's an entry in Function.json? So for IBinder scenarios, it's skipped.  ([https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Binding/ApiHubScriptBindingProvider.cs#L105](https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Binding/ApiHubScriptBindingProvider.cs#L105) )